db/migrate/20220512123911_create_request_logs.rb

```diff
@@ -0,0 +1,11 @@
+class CreateRequestLogs < ActiveRecord::Migration[6.0]
+  def change
+    create_table :request_logs do |t|
+      t.references :miq_requests, :type => :bigint
+      t.string :log_message
+      t.string :object_type
+      t.bigint :object_id
```

Comment:
3 comments here:

---

@akhilkr128 What is an `object` in this context?  I'm worried that this is too generic of a term.

---

Think this can be simplified to `t.references :object, :polymorphic => true`, but you'd have to verify.  (@agrare Do you know?)

---

Additionally, I'm wondering if this should have an index on it.

Reply:
Yes this is the normal pattern we use: `t.references :resource, :type => :bigint, :polymorphic => true`

Reply:
Ah thanks I missed the type - I'll update my comment / suggestion.

Reply:
I planned to give all the below values as strings to object type.
'ServiceAnsiblePlaybook'
'ServiceTemplateProvisionTask'
'ServiceTemplateProvisionRequest'
'ServiceTemplate'
'AutomationTask'
'AutomationRequest'

Reply:
is that different from the request's type?

Reply:
I will try for the polymorphic relation. Will get back.

Reply:
@akhilkr128 We had discussed maybe replacing the object_type/id with an miq_request_task_id, because I thought it was only MiqRequestTask and MiqRequest values (and we already had an miq_request_id).

However, now I see that this column could contain a number of different objects, namely: Service, MiqRequestTask, MiqRequest, and ServiceTemplate.  So, instead I think we should drop miq_request_id in favor of the polymorphic object_type/object_id.

Only other thing I would say here is use the term `resource` instead of `object`, since that's more consistent with the rest of the code base.

Reply:
Sure, make sense.